File: backend/src/main/java/terminal_code/backend/entity/Workspace.java
package terminal_code.backend.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "workspaces")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Workspace {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(nullable = false)
    private String name;

    private String description;
    private String color;
    private String icon;

    @Column(name = "is_starred", nullable = false)
    private boolean isStarred;

    @Column(name = "sort_order", nullable = false)
    private int sortOrder;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    void prePersist() {
        Instant now = Instant.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    void preUpdate() {
        this.updatedAt = Instant.now();
    }

    public Workspace(UUID userId, String name, String description, String color, String icon) {
        this.userId = userId;
        this.name = name;
        this.description = description;
        this.color = color;
        this.icon = icon;
    }
}

File: frontend/src/types/workspace.ts
export interface Project {
  id: string;
  name: string;
  localPath: string;
  description: string | null;
  isStarred: boolean;
  lastOpenedAt: string | null;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
}

export interface Workspace {
  id: string;
  name: string;
  description: string | null;
  color: string | null;
  icon: string | null;
  isStarred: boolean;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
  projects: Project[];
}

export interface CreateWorkspaceRequest {
  name: string;
  description?: string;
  color?: string;
  icon?: string;
}

export interface UpdateWorkspaceRequest {
  name?: string;
  description?: string;
  color?: string;
  icon?: string;
  isStarred?: boolean;
  sortOrder?: number;
}

export interface CreateProjectRequest {
  name: string;
  localPath: string;
  description?: string;
}
